timeseries-bucket-repair/timeseries_bucket_repair.js

```diff
@@ -0,0 +1,150 @@
+// ------------------------------------------------------------------------------------
+// This script provides functionality to unconditionally re-write an internal time-series bucket.
+// It can be used for most types of logical bucket corruption detected by validate.
+// ------------------------------------------------------------------------------------
+
+//
+// Helper function to validate namespaces, create temporary collection and return our timeseries
+// options.
+//
+function verifyAndSetupCollsAndGetTSOptions(collName, tempColl) {
```

Comment:
This is a good check that should be added to the other support-tools scripts that have the same pattern of creating a temp collection and then re-inserting into original collection as a transaction. I filed [SERVER-106642](https://jira.mongodb.org/browse/SERVER-106642)

Reply:
Cool, thanks!